src/core/components/hv-route/index.tsx

```diff
@@ -313,6 +323,36 @@ class HvRouteInner extends PureComponent<Types.InnerRouteProps, Types.State> {
   }
 }
 
+/**
+ * Retrieve a nested navigator as a child of the nav-route with the given id
+ */
+const getNestedNavigator = (
+  id?: string,
+  doc?: TypesLegacy.Document,
+): TypesLegacy.Element | undefined => {
+  if (!id || !doc) {
+    return undefined;
+  }
+
+  const routes = doc
+    .getElementsByTagNameNS(
+      Namespaces.HYPERVIEW,
+      TypesLegacy.LOCAL_NAME.NAV_ROUTE,
+    )
+    .filter((n: TypesLegacy.Element) => {
+      return n.getAttribute('id') === id;
+    });
+  const route = routes && routes.length > 0 ? routes[0] : undefined;
```

Comment:
I think this is the same with a little less iterations:
```suggestion
  const route = doc.getElementById(id)
```

Reply:
This is what I first tried but this created a difference between document and element which made a bunch of other TypeScript calls (like getFirstTag) no longer work. This was the quickest way to move forward.